lib/factory_factory_girl/generators/model_generator.rb

```diff
@@ -18,21 +18,19 @@ def factory_attributes
       end
 
       def set_column(attribute)
-        match_results = rules.map do |rule|
-          if attribute.name.match(rule[:rule])
-            rule
-          end
-        end.compact
+        applied_rule = get_match_results(rules, attribute).first
+        return default_value(attribute) unless applied_rule
+        return "{ #{applied_rule[:function]} }" unless applied_rule[:value]
```

Comment:
I think use `if` statement here would be more straightforward. 
```
if applied_rule[:value]
  transfer_value_type(applied_rule[:value], attribute.type.to_s)
else
  "{ #{applied_rule[:function]} }"
end
```

But extract the default_value condition is a good idea.😀